src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -275,13 +319,11 @@ private static (SyntaxTriviaList sameLine, SyntaxTriviaList endOfLine) SplitTrai
                     continue;
                 }
 
-                if (ContainsReference(childNode, localVariable, semanticModel, syntaxFactsService, cancellationToken))
-                {
-                    return true;
-                }
+                var variables = GetReferencedLocalVariables(childNode, localVariables, semanticModel, syntaxFactsService, cancellationToken);
+                referencedVariables.AddRange(variables);
```

Comment:
each recursive call will allocate an array.  Let's avoid that.  just create the set up front (perhaps in the caller) and pass into this to be fille.d